lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -1,26 +1,37 @@
 'use strict';
 
 const _ = require('lodash');
+const BbPromise = require('bluebird');
+const { addCustomResourceToService } = require('../../../../customResources');
 
 class AwsCompileS3Events {
-  constructor(serverless) {
+  constructor(serverless, options) {
     this.serverless = serverless;
+    this.options = options;
     this.provider = this.serverless.getProvider('aws');
 
     this.hooks = {
-      'package:compileEvents': this.compileS3Events.bind(this),
+      'package:compileEvents': () => {
+        return BbPromise.bind(this)
+          .then(this.newS3Buckets)
+          .then(this.existingS3Buckets);
+      },
     };
   }
 
-  compileS3Events() {
+  newS3Buckets() {
     const bucketsLambdaConfigurations = {};
     const s3EnabledFunctions = [];
     this.serverless.service.getAllFunctions().forEach(functionName => {
       const functionObj = this.serverless.service.getFunction(functionName);
 
       if (functionObj.events) {
+        // eslint-disable-next-line consistent-return
```

Comment:
I don't think this disable is needed. At below line we should just do `return;` instead of `return null`, and there shold be no error reported by linter.

Reply:
I can remove this line but we still need to return `null` (not `return;`). Otherwise we get the same consistent return violation error.

Reply:
Ok, good, my point was, that it's best if we rely on `eslint-disable-next-line` only where there's absolutely no way to go without it (and we want to keep rule on, for some good reasons).

If `return null` is the way, then it's totally fine :)